cloudwatch_monitoring/sns.py

```diff
@@ -0,0 +1,184 @@
+        'width': positioning['width'],
+        'properties': {
+            "metrics": [
+                ["AWS/SNS", "NumberOfMessagesPublished", "TopicName", f"aqts-capture-error-handler-{deploy_stage}-topic", {"label": "messages published"}],
+                [".", "NumberOfNotificationsDelivered", ".", ".", {"label": "notifications delivered"}],
+                [".", "NumberOfNotificationsFailed", ".", ".", {"label": "notifications failed"}]
+            ],
+            "view": "timeSeries",
+            "stacked": False,
+            "region": region,
+            "stat": "Sum",
+            "period": 3600,
+            "title": "Error Handler Failure Notifications (Counts per hour)"
+        }
+    }
+
+    sns_widgets.append(error_handler_sns_notifications_widget)
+
+    # create the template for the generic sns notifications widget
+    sns_notifications_widget = {
+        'type': 'metric',
+        'height': positioning['height'],
+        'width': positioning['width'],
+        'properties': {
+            "metrics": [],
+            "view": "timeSeries",
+            "stacked": False,
+            "region": region,
+            "stat": "Sum",
+            "period": 3600,
+            "title": "SNS Failure Notifications (Count per hour)"
+        }
+    }
+
+    api_calls = SNSAPICalls(region, deploy_stage)
+
+    # grab all the sns topics in the account/region
+    response = api_calls.get_all_sns_topics()
+
+    # iterate over the list of sns topics and create widgets for the assets we care about based on filters
+    count = 0
+    for topic in response['Topics']:
+        topic_arn = topic['TopicArn']
+        if api_calls.is_iow_topic_filter(topic_arn):
+            # incoming topic arn example: arn:aws:sns:us-west-2:579777464052:aqts-capture-error-handler-DEV-topic
+            # we want the topic name after the last ":"
+            arn_parts = topic_arn.split(':')
+            topic_name = arn_parts[-1]
+            tier_agnostic_topic_name = topic_name.replace(f"-{deploy_stage}-topic", '')
+
+            metric = generate_number_of_messages_published_metric(topic_name, tier_agnostic_topic_name, count)
+            sns_notifications_widget['properties']['metrics'].append(metric)
+            count += 1
+
+    sns_widgets.append(sns_notifications_widget)
+
+    return sns_widgets
+
+
+def generate_number_of_messages_published_metric(topic_name, tier_agnostic_topic_name, count):
+    try:
+        label = sns_topics[tier_agnostic_topic_name]['title']
+    except KeyError:
+        # No title property or possibly lookup added for this sns topic yet, default to topic name
+        label = topic_name
+
+    if count < 1:
+        # the first metric in the list has some additional stuff
+        metric = ["AWS/SNS", 'NumberOfMessagesPublished', "TopicName", topic_name, {"label": label}]
+    else:
+        metric = ["...", topic_name, {"label": label}]
+
+    return metric
+
+
+class SNSAPICalls:
```

Comment:
Does this violate camel case? Should it be SnsApiCalls?  Don't know.

Reply:
Probably, I will rename, I think I do this with the SQS api calls too....